api/tests/unit/domain/usecases/get-session-for-supervising_test.js

```diff
@@ -1,111 +1,193 @@
 const { expect, sinon, domainBuilder } = require('../../../test-helper');
 const getSessionForSupervising = require('../../../../lib/domain/usecases/get-session-for-supervising');
+const { constants } = require('../../../../lib/domain/constants');
+const moment = require('moment');
+
+const START_DATETIME_STUB = new Date('2022-10-01T13:00:00Z');
+const sessionForSupervisingRepository = { get: sinon.stub() };
+
+const expectedSessionEndDateTimeFromStartDateTime = (startDateTime, addedTime) => {
+  return moment(startDateTime).add(addedTime, 'minutes').toDate();
+};
 
 describe('Unit | UseCase | get-session-for-supervising', function () {
+  afterEach(function () {
+    sinon.reset();
+  });
+
   context('when the session exists', function () {
     it('should fetch and return the session from repository', async function () {
       // given
       const expectedSession = domainBuilder.buildSessionForSupervising();
-      const sessionForSupervisingRepository = { get: sinon.stub() };
       sessionForSupervisingRepository.get.resolves(expectedSession);
 
       // when
       const actualSession = await getSessionForSupervising({ sessionId: 1, sessionForSupervisingRepository });
 
       // then
-      expect(actualSession).to.equal(expectedSession);
+      expect(actualSession).to.deep.equal(expectedSession);
     });
 
     context('when there are candidates', function () {
-      context('when some candidates are still eligible to complementary certifications', function () {
-        it("should return the session with the candidates' eligibility", async function () {
+      context('when some candidates have no complementary certifications', function () {
+        it('should not compute a theorical end datetime if session has not started', async function () {
```

Comment:
Nitpick: "... if certification has not started yet"

Reply:
Modifié!